prime-router/docs/design/validation/design.md

```diff
@@ -0,0 +1,257 @@
+CachingValidationSupport cache = new CachingValidationSupport(supportChain);
+
+// Create a validator using the FhirInstanceValidator module. We can use this
+// validator to perform validation
+FhirInstanceValidator validatorModule = new FhirInstanceValidator(cache);
+FhirValidator validator = ctx.newValidator().registerValidatorModule(validatorModule);
+ValidationResult result = validator.validateWithResult(input);
+```
+</details>
+
+#### Lots more information [here](https://hapifhir.io/hapi-fhir/docs/validation/introduction.html)
+
+## Resulting Tickets
+
+- Implement reusable validation component [#8974](https://github.com/CDCgov/prime-reportstream/issues/8974)
+- Research validation configurations: [#8976](https://github.com/CDCgov/prime-reportstream/issues/8976)
+- Add FHIR Sender Converter Validation: [#8978](https://github.com/CDCgov/prime-reportstream/issues/8978)
+- Add FHIR Internal Validation:  [#8979](https://github.com/CDCgov/prime-reportstream/issues/8979)
+- Enable Parser validation and check result: [#8980](https://github.com/CDCgov/prime-reportstream/issues/8980)
+- Configure FHIR parser, sender, internal, and receiver validation profiles: [#9034](https://github.com/CDCgov/prime-reportstream/issues/9034)
+- Add FHIR Receiver Validation: [#9035](https://github.com/CDCgov/prime-reportstream/issues/9035)
+- Test default validation against existing messages: https://github.com/CDCgov/prime-reportstream/pull/9165
+
+## HL7 Validation
+### Proposed Design
+The diagram below proposes two validation "checkpoints":
+1. Sender validation - will occur during the convert function, after receiving and debatch, and before the convert step.
+2. Receiver Validation - will occur in the translate function, after reception, translation, and enrichment but right
+   before dispatch.
+
+We want to add sender validation so that we can ensure that the data we are receiving matches the intended spec so that
+we do not send messy data.
+We want to add receiver validation so that the receiver can receive the data in the format they are expecting.  
+
+![hl7-validation-annotated-architecture-diagram.png](annotated-hl7-architecture-diagram.png)
+
+### Background Information/Dev Notes
+#### RADx MARS
+It is important that we also take into consideration waht MARS expects of us when it comes to validation:
+<details>
+<summary>MARS Product Requirements for Hubs</summary>
+ 
+The purpose of this document is to outline requirements for how ReportStream (and
+other hubs) should handle HL7v2 files received by senders who conform to the
+MARS specification.
+Upon receiving an HL7v2 file from a MARS sender, ReportStream should:
+1. Perform in-line validation of the message
+2. Route the HL7v2 file to the appropriate state/local jurisdiction, based on pre-
+   defined state preferences for receiving OTC data
+3. Strip PII from the HL7v2 file and route it to HHS Protect
+   Each step is described in more detail below.
+1. Perform in-line validation of the message
```

Comment:
@thetaurean I don't think this is accurate. There isn't a NIST validator to just call that I can find, unless you found something?

Reply:
Can we resolve this? @JessicaWNava